ydb/docs/en/core/concepts/federated_query/greenplum.md

```diff
@@ -0,0 +1,76 @@
+# Working with Greenplum databases
+
+This section provides basic information on working with external [Greenplum](https://greenplum.org) databases.
+
+To work with an external Greenplum database, you need to follow these steps:
+1. Create a [secret](../datamodel/secrets.md) containing the password for connecting to the database.
+    ```sql
+    CREATE OBJECT greenplum_datasource_user_password (TYPE SECRET) WITH (value = "<password>");
+    ```
+1. Create an [external data source](../datamodel/external_data_source.md) that describes a specific database within the Greenplum cluster. In the `LOCATION` parameter you need to pass the network address of [master node](https://www.greenplumdba.com/greenplum-master) Greenplum. By default, the [namespace](https://docs.vmware.com/en/VMware-Greenplum/6/greenplum-database/ref_guide-system_catalogs-pg_namespace.html) `public` is used for reading, but this value can be changed using the optional `SCHEMA` parameter. The network connection is made using the standard ([Frontend/Backend Protocol](https://www.postgresql.org/docs/current/protocol.html)) over TCP transport (`PROTOCOL="NATIVE"`). You can enable encryption of connections to the external database using the `USE_TLS="TRUE"` parameter.
+    ```sql
+    CREATE EXTERNAL DATA SOURCE greenplum_datasource WITH (
+        SOURCE_TYPE="Greenplum",
+        LOCATION="<host>:<port>",
+        DATABASE_NAME="<database>",
+        AUTH_METHOD="BASIC",
+        LOGIN="user",
+        PASSWORD_SECRET_NAME="greenplum_datasource_user_password",
+        PROTOCOL="NATIVE",
+        USE_TLS="TRUE",
+        SCHEMA="<schema>"
+    );
+    ```
+1. {% include [!](_includes/connector_deployment.md) %}
+1. [Execute a query](#query) to the database.
+
+## Query syntax { #query }
+The following SQL query format is used to work with Greenplum:
+
+```sql
+SELECT * FROM greenplum_datasource.<table_name>
+```
+
+where:
+- `greenplum_datasource` - identifier of the external data source;
+- `<table_name>` - table name within the external data source.
+
+## Limitations
+
+When working with Greenplum clusters, there are a number of limitations:
+
+1. {% include [!](_includes/supported_requests.md) %}
+1. {% include [!](_includes/datetime_limits.md) %}
+1. {% include [!](_includes/predicate_pushdown.md) %}
+
+## Supported data types
+
+In the Greenplum database, the optionality of column values (whether a column can contain `NULL` values) is not part of the data type system. The `NOT NULL` constraint for each column is implemented as the `attnotnull` attribute in the system catalog [pg_attribute](https://www.postgresql.org/docs/current/catalog-pg-attribute.html), i.e., at the metadata level of the table. Therefore, all basic PostgreSQL types can contain `NULL` values by default, and in the {{ ydb-full-name }} type system, they should be mapped to [optional](https://ydb.tech/docs/ru/yql/reference/types/optional) types.
```

Comment:
Let's explicitly mention in the introduction that Greenplum is based on PostgreSQL, so it doesn't feel like a surprise when PostgreSQL is mentioned in the middle of the article. Also, a lot of this looks like copy-paste, which better be moved to includes for deduplication.